### PR TITLE
feat: Setting up the project for web SDK

### PR DESCRIPTION
**Pull request by hiranya911**
Setting up a new web SDK project with:
* TypeScript,
* API Extractor, and
* Mocha unit tests.

Formatted the code with prettier since the repo is already configured to use it. The existing packages in this repo do not have any tests at the moment. In order to run unit tests only on the newly added package, I've configured a separate Actions workflow.

### CODE REVIEW

File: firestore-stripe-web-sdk/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "firestore-stripe-payments-js",
+  "version": "0.0.1",
+  "description": "Client SDK for the firestore-stripe-payments Firebase Extension",
+  "main": "./lib/index.js",
+  "typings": "./lib/index.d.ts",
+  "scripts": {
+    "api-extractor:local": "npm run build && api-extractor run --local --verbose",
```
Comment (Feiyang1) [inline firestore-stripe-web-sdk/package.json:8]: You may want to set up a workflow to update the API report to make sure it's always up to date in PRs. https://github.com/firebase/firebase-js-sdk/blob/master/.github/workflows/update-api-reports.yml

File: firestore-stripe-web-sdk/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "api-extractor": "api-extractor run --verbose",
```
Comment (Feiyang1) [inline firestore-stripe-web-sdk/package.json:9]: How are you going to generate ref docs?

**Review by Feiyang1 — APPROVED** [review]

File: firestore-stripe-web-sdk/package.json (hunk 1 above)
Comment (hiranya911) [reply firestore-stripe-web-sdk/package.json:8]: With the current setup, the PR will fail to build if the API report is not up-to-date. PR authors must run `api-extractor:local` and commit the changes as part of the PR to get a successful build.

File: firestore-stripe-web-sdk/package.json (hunk 2 above)
Comment (hiranya911) [reply firestore-stripe-web-sdk/package.json:9]: Probably API Documenter. We can add the necessary tooling in a future PR.

**Pull request merged**